src/main/java/put/io/black/java/ScenarioGUI.java

```diff
@@ -20,13 +31,61 @@
     private JScrollPane inputFieldScroll;
     private JScrollPane outputFieldScroll;
 
+    private String localhost = "http://localhost:8080/";
+
+    private static final Logger logger = LoggerFactory.getLogger(TextTransformerController.class);
+
+    /**
+     * Class to handle listener from GUI.
+     */
     public ScenarioGUI() {
         howManyStepsScenario.addMouseListener(new MouseAdapter() {
             @Override
             public void mouseClicked(MouseEvent e) {
                 super.mouseClicked(e);
-                //TODO API liczba krokow w scenariuszu
-                outputField.setText("Dużo kroków w scenariuszu!");
+                //TODO API poprawna funkcje wlozyc (teraz jest stnadard co bylo w projektcie z moodle)
+                if (inputField.getText() == "") {
+                    JOptionPane.showMessageDialog(null, "Pole wejściowe scenariusza jest puste!");
+                    logger.error("Pole wejściowe scenariusza jest puste!");
+                } else {
+                    try {
+                        String prepareUrl = localhost + inputField.getText();
+                        String correctUrl = stringToUrl(prepareUrl);
+                        URL url = new URL(correctUrl);
+                        logger.debug(url.toString());
+                        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
```

Comment:
Connect to server, send data and receive response should be placed in the external class. Now in this class, you have all steps. Prepare URI builder module, connection with server method and use it as 'your API'.